crates/rpc_client/src/lib.rs

```diff
@@ -32,25 +46,31 @@ pub fn get_tx_by_hash(
             ),
         )))
         .send()
-        .expect("failed to send");
+        .map_err(|err| SendError {
+            msg: err.to_string(),
+        })?
+        .text()
+        .map_err(|err| ResponseError {
+            msg: err.to_string(),
+        })?;
 
-    let response_text_ = response.text().expect("failed to get response text");
-    let response_text = response_text_.as_str();
+    let response_text_str = response_text.as_str();
 
-    let success: jsonrpc_types::Success<Transaction> = serde_json::from_str(response_text)
-        .unwrap_or_else(|_| {
-            panic!(
-                "failed to interpret as Transaction response string \"{}\"",
-                response_text
-            )
-        });
+    let success: jsonrpc_types::Success<Transaction> = serde_json::from_str(response_text_str)
```

Comment:
I think you can do `serde_json::From_str(&response_text)` instead. `&` can often be used to reference a value as a slice type

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3383/commits/14ef314e20a428ea4f5f19a79b97ede55e27ed14